cpp/arcticdb/async/async_store.hpp

```diff
@@ -22,227 +22,263 @@ namespace arcticdb::async {
 template<class ClockType = util::SysClock>
 class AsyncStore : public Store {
 public:
-    AsyncStore(
-        std::shared_ptr<storage::Library> library,
-        const arcticdb::proto::encoding::VariantCodec &codec) :
-        library_(std::move(library)),
-        codec_(new arcticdb::proto::encoding::VariantCodec{codec}) {
+    AsyncStore(std::shared_ptr<storage::Library> library, const arcticdb::proto::encoding::VariantCodec& codec)
+        : library_(std::move(library)),
+          codec_(new arcticdb::proto::encoding::VariantCodec{codec})
+    {
     }
 
-    folly::Future<entity::VariantKey> write(
-        stream::KeyType key_type,
+    folly::Future<entity::VariantKey> write(stream::KeyType key_type,
         VersionId version_id,
         const StreamId& stream_id,
         IndexValue start_index,
         IndexValue end_index,
-        SegmentInMemory &&segment) override {
+        SegmentInMemory&& segment) override
+    {
 
         util::check(segment.descriptor().id() == stream_id,
-                    "Descriptor id mismatch in atom key {} != {}",
-                    stream_id,
-                    segment.descriptor().id());
-
-        return async::submit_cpu_task(EncodeAtomTask{
-            key_type, version_id, stream_id, start_index, end_index, current_timestamp(),
-            std::move(segment), library_, codec_, size_t(0)
-        })
+            "Descriptor id mismatch in atom key {} != {}",
+            stream_id,
+            segment.descriptor().id());
+
+        return async::submit_cpu_task(EncodeAtomTask{key_type,
+                                          version_id,
+                                          stream_id,
+                                          start_index,
+                                          end_index,
+                                          current_timestamp(),
+                                          std::move(segment),
+                                          library_,
+                                          codec_,
+                                          size_t(0)})
             .via(&async::io_executor())
             .thenValue(WriteSegmentTask{library_});
     }
 
-    folly::Future<entity::VariantKey> write(
-            stream::KeyType key_type,
-            VersionId version_id,
-            const StreamId& stream_id,
-            timestamp creation_ts,
-            IndexValue start_index,
-            IndexValue end_index,
-            SegmentInMemory &&segment) override {
+    folly::Future<entity::VariantKey> write(stream::KeyType key_type,
+        VersionId version_id,
+        const StreamId& stream_id,
+        timestamp creation_ts,
+        IndexValue start_index,
+        IndexValue end_index,
+        SegmentInMemory&& segment) override
```

Comment:
When Egpyption brackets are used, the argument list needs extra indentation to offset it from the function body.